docs/api.rst

```diff
@@ -11486,3 +11486,174 @@ Handling user notes
    :statuscode 400: Provided JSON is in some way malformed.
    :statuscode 409: No user is currently logged in. User note with the given title already exists. User has reached the limit of available notes. Check error message.
    :statuscode 500: Internal rotki error.
+
+
+Custom Assets
+================
+
+.. http:post:: /api/(version)/assets/custom
+
+   Doing a POST on this endpoint will return all the custom assets present in the database.
+
+   **Example Request**:
+
+   .. http:example:: curl wget httpie python-requests
+
+      POST /api/1/assets/custom HTTP/1.1
+      Host: localhost:5042
+      Content-Type: application/json;charset=UTF-8
+
+      {"name": "land", "custom_asset_type": "real estate"}
+
+   :reqjson int limit: This signifies the limit of records to return as per the `sql spec <https://www.sqlite.org/lang_select.html#limitoffset>`__.
+   :reqjson int offset: This signifies the offset from which to start the return of records per the `sql spec <https://www.sqlite.org/lang_select.html#limitoffset>`__.
+   :reqjson string[optional] name: The name of the custom asset to be used as a filter.
+   :reqjson string[optional] custom_asset_type: The type of the custom asset to be used as a filter.
+
+   **Example Response**:
+
+   .. sourcecode:: http
+
+      HTTP/1.1 200 OK
+      Content-Type: application/json
+
+      {
+          "result": {
+                "entries": [{
+                        "identifier": "00fe5f97-882b-42e5-b0e0-39ebd3a99156'",
+                        "name": "House",
+                        "notes": "Apartment purchase",
+                        "custom_asset_type": "real estate",
+                    },
+                    {
+                        "identifier": "333c248f-7c64-41f4-833b-2fad96c4ea6b",
+                        "name": "Ferrari",
+                        "notes": "Exotic car inheritance from lineage",
+                        "custom_asset_types": "vehicles",
+                }],
+                "entries_found": 2,
+                "entries_total": 2,
+                "entries_limit": -1
+            },
+          "message": ""
+      }
+
+   :resjson object result: An array of objects representing custom assets entries.
+   :resjson int identifier: The unique identifier of the custom asset.
+   :resjson str name: The name of the custom asset.
+   :resjson str notes: The notes used as a description of the custom asset.
+   :resjson str custom_asset_type: The type/category of the custom asset.
+   :resjson int entries_found: The number of entries found for the current filter. Ignores pagination.
+   :resjson int entries_limit: The limit of entries. Constant value of -1.
```

Comment:
This is used in every location that pagination is used. ETH txn, etc.

Reply:
Ohhh okay. So basically the whole idea/reason behind it is for the front end to handle pagination quite nicely.

`limit` & `offset` coupled with `entries_limit` help to handle the freemium model of specific endpoints i.e.
`Initial request -> limit=10&offset=0; response -> {'entries': [], 'entries_limit': 100, 'entries_found': 400, 'entries_total': 1000`

With that info, the FE knows that there's a limit of 100 items that can be paginated out of 400 items that match that query whereby the entire table has 1k items/rows.
So subsequent requests will look like this `limit=10&offset=10` since we got the first 10 items.

PS: The `entries_*` keys are also used to generate the page number i.e `1, 2, 3` given that there's a known increment and a total number of items that can be fetched.

Reply:
Yeah, thanks, I understand the purpose of the pagination-related variables but I mean what is the point of returning constant `-1` ? Maybe we should not return it at all?